src/core/algorithms/nd/nd_verifier/nd_verifier.h

```diff
@@ -0,0 +1,82 @@
+#pragma once
+
+#include <unordered_map>
+#include <unordered_set>
+#include <utility>
+#include <vector>
+
+#include "algorithms/algorithm.h"
+#include "algorithms/nd/nd.h"
+#include "algorithms/nd/nd_verifier/util/stats_calculator.h"
+#include "config/equal_nulls/type.h"
+#include "config/indices/type.h"
+#include "config/tabular_data/input_table_type.h"
+#include "model/table/column_layout_relation_data.h"
+#include "model/table/column_layout_typed_relation_data.h"
+
+namespace algos::nd_verifier {
+
+using namespace algos::nd_verifier::util;
+
+/// @brief Algorithm for verifying if ND holds with given weight
+class NDVerifier : public Algorithm {
+private:
+    config::InputTable input_table_;
+    config::IndicesType lhs_indices_;
+    config::IndicesType rhs_indices_;
+    model::WeightType weight_;
+    config::EqNullsType is_null_equal_null_;
+
+    using ValueType = std::string;
+    using CodeType = size_t;
```

Comment:
I haven't found any use for these usings. If it is really not used, it should be removed.